File: isegoria/src/main/java/com/eulersbridge/isegoria/models/Event.java
package com.eulersbridge.isegoria.models;

import android.support.annotation.Nullable;

import com.eulersbridge.isegoria.network.Timestamp;
import com.squareup.moshi.Json;

import java.util.List;

@org.parceler.Parcel
public class Event {

    public String name;
    public String location;
    public String description;

    public String organizer;
    public String organizerEmail;

    @Json(name = "created")
    @Timestamp
    public long date;

    public List<Photo> photos;

    public @Nullable String getPhotoUrl() {
        if (photos != null && photos.size() > 0) {
            return photos.get(0).thumbnailUrl;

        } else {
            return null;
        }
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/Event.kt
package com.eulersbridge.isegoria.network.api.models

import android.annotation.SuppressLint
import android.os.Parcelable
import com.eulersbridge.isegoria.network.adapters.Timestamp
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@SuppressLint("ParcelCreator")
@Parcelize
data class Event(
    val name: String?,
    val location: String?,
    val description: String?,
    val organizer: String?,
    val organizerEmail: String?,
    @Json(name = "created") @Timestamp val date: Long = 0,
    val photos: List<Photo>?
) : Parcelable {

    fun getPhotoUrl(): String? {
        return if (photos != null && photos.isNotEmpty()) {
            photos[0].thumbnailUrl
        } else {
            null
        }
    }
}